tests/unit/s2n_extension_type_lists_test.c

```diff
@@ -19,108 +19,32 @@
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_SAME_IN_TLS13, conn, &default_list));
-            EXPECT_NOT_EMPTY_LIST(default_list);
-
-            conn->actual_protocol_version = S2N_TLS13;
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_SAME_IN_TLS13, conn, &tls13_list));
-            EXPECT_NOT_EMPTY_LIST(tls13_list);
-
-            EXPECT_EQUAL(default_list->extension_types, tls13_list->extension_types);
-        }
-
-        /* Can retrieve different lists for tls1.2 and tls1.3 */
-        {
-            s2n_extension_type_list *default_list = NULL, *tls13_list = NULL;
+        s2n_extension_type_list *list = NULL;
 
-            conn->actual_protocol_version = S2N_TLS12;
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_DIFFERENT_IN_TLS13, conn, &default_list));
-            EXPECT_NOT_EMPTY_LIST(default_list);
-
-            conn->actual_protocol_version = S2N_TLS13;
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_DIFFERENT_IN_TLS13, conn, &tls13_list));
-            EXPECT_NOT_EMPTY_LIST(tls13_list);
-
-            EXPECT_NOT_EQUAL(default_list->extension_types, tls13_list->extension_types);
-        }
-
-        /* Retrieves default list when protocol version earlier than tls1.2 */
+        /* Safety checks */
         {
-            s2n_extension_type_list *default_list = NULL, *tls10_list = NULL;
-            conn->actual_protocol_version = S2N_TLS12;
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_DIFFERENT_IN_TLS13, conn, &default_list));
-            EXPECT_NOT_EMPTY_LIST(default_list);
+            EXPECT_FAILURE(s2n_extension_type_list_get(0, NULL));
 
-            conn->actual_protocol_version = S2N_TLS10;
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_DIFFERENT_IN_TLS13, conn, &tls10_list));
-            EXPECT_NOT_EMPTY_LIST(tls10_list);
-
-            EXPECT_EQUAL(default_list->extension_types, tls10_list->extension_types);
+            /* Should fail for a bad list type */
+            EXPECT_FAILURE(s2n_extension_type_list_get(-1, &list));
+            EXPECT_FAILURE(s2n_extension_type_list_get(S2N_EXTENSION_LIST_IDS_COUNT, &list));
         }
 
-        /* Can retrieve an empty list */
-        {
-            s2n_extension_type_list *empty_list = NULL;
-            conn->actual_protocol_version = S2N_TLS12;
-            EXPECT_SUCCESS(s2n_extension_type_list_get(LIST_TYPE_EMPTY_IN_TLS12, conn, &empty_list));
-            EXPECT_NOT_NULL(empty_list);
-            EXPECT_EQUAL(empty_list->count, 0);
-            EXPECT_NULL(empty_list->extension_types);
-        }
-
-        /* Fails to retrieve an invalid list id */
-        {
-            conn->actual_protocol_version = S2N_TLS12;
-            EXPECT_FAILURE(s2n_extension_type_list_get(S2N_EXTENSION_LIST_IDS_COUNT, conn, &list));
-        }
-
-        /* Can retrieve a list for every id + protocol version */
+        /* Can retrieve a list for every id */
         {
```

Comment:
Why remove the check for  every id + protocol version?

Reply:
Because the function no longer takes `conn` or considers protocol version